feat: add `lapack/base/dlae2` (#2824)
stdlib-js/stdlib: lib/node_modules/@stdlib/lapack/base/dlae2/README.md

<!--

@license Apache-2.0

Copyright (c) 2024 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# dlae2

> Compute the eigenvalues of a 2x2 symmetric matrix.

<section class = "usage">

## Usage

```javascript
var dlae2 = require( '@stdlib/lapack/base/dlae2' );
```

> #### dlae2( A, B, C, Out )

Reviewer: ```suggestion
#### dlae2( A, B, C, out )
```

You can lowercase `out` here and below. We use `out` elsewhere in the project and it is always lowercased.

Computes the eigenvalues of a 2x2 symmetric matrix.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var out = new Float64Array( 2 );
out = dlae2( 2.0, 3.0, 4.0, out );
// out => <Float64Array>[ 6.16227766016838, -0.1622776601683793 ]
```

The function has the following parameters:

-   **A**: the (0,0) element of matrix.

> -   **B**: the (0,1) & (1,0) elements of matrix.

Reviewer: ```suggestion
-   **B**: the (0,1) and (1,0) elements of matrix.
```

Don't use symbol abbreviations in prose.

-   **C**: the (1,1) element of matrix.

> -   **Out**: output [`Float64Array`][mdn-float64array] containing eigenvalue of larger and smaller absolute value respectively.

Reviewer: ```suggestion
-   **Out**: output [`Float64Array`][mdn-float64array] containing the eigenvalues of the larger and smaller absolute values, respectively.
```

Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.

<!-- eslint-disable stdlib/capitalized-comments -->

```javascript
var Float64Array = require( '@stdlib/array/float64' );

> // Initial arrays...

Reviewer: ```suggestion
// Initial array:
```

> var OUT0 = new Float64Array( [ 0.0, 0.0, 0.0 ] );

Reviewer: ```suggestion
var out0 = new Float64Array( [ 0.0, 0.0, 0.0 ] );
```

No all caps for `out` here and below. This is just a matter of convention.

> // Create offset views...

Reviewer: ```suggestion
// Create an offset view...
```

Only one array is created here.

var OUT1 = new Float64Array( OUT0.buffer, OUT0.BYTES_PER_ELEMENT*1 ); // start at 2nd element

dlae2( 2.0, 3.0, 4.0, OUT1 );
// OUT0 => <Float64Array>[ 0.0, 6.16227766016838, -0.1622776601683793 ]
```

#### dlae2.ndarray( A, B, C, Out, strideOut, offsetOut )

Computes the eigenvalues of a 2x2 symmetric matrix using alternative indexing semantics.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var out = new Float64Array( 2 );
out = dlae2.ndarray( 2.0, 3.0, 4.0, out, 1, 0 );

> // out => <Float64Array>[ 6.16227766016838, -0.1622776601683793 ]

Reviewer: Use approximate values, rather than exact, lest the implementation change and thus the precision of the returned results.

```

The function has the following additional parameters:

-   **strideOut**: stride length for `Out`.
-   **offsetOut**: starting index of `out`

While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameters support indexing semantics based on starting indices. For example,

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var out = new Float64Array( [ 0.0, 0.0, 0.0 ] );
out = dlae2.ndarray( 2.0, 3.0, 4.0, out, 1, 1 );
// out => <Float64Array>[ 0.0, 6.16227766016838, -0.1622776601683793 ]
```

</section>

<!-- /.usage -->

<section class="notes">

## Notes

-   `dlae2()` corresponds to the [LAPACK][lapack] routine [`dlae2`][lapack-dlae2].

</section>

<!-- /.notes -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Float64Array = require( '@stdlib/array/float64' );
var dlae2 = require( '@stdlib/lapack/base/dlae2' );

var out = new Float64Array( 2 );

> out = dlae2( 2.0, 3.0, 4.0, out );

Reviewer: This is effectively the same example as above. I suggest being more creative.
